nats_test.go

```diff
@@ -98,6 +98,94 @@ func TestVersionMatchesTag(t *testing.T) {
 	}
 }
 
+func TestExpandTildePath(t *testing.T) {
+	t.Run("unix", func(t *testing.T) {
+		if runtime.GOOS == "windows" {
+			t.Skip()
+			return
+		}
+
+		origHome := os.Getenv("HOME")
+		defer os.Setenv("HOME", origHome)
+
+		cases := []struct {
+			path string
+			home string
+
+			wantPath string
+			wantErr  bool
+		}{
+			{path: "/foo/bar", home: "/fizz/buzz", wantPath: "/foo/bar", wantErr: false},
+			{path: "foo/bar", home: "/fizz/buzz", wantPath: "foo/bar", wantErr: false},
+			{path: "~/fizz", home: "/foo/bar", wantPath: "/foo/bar/fizz", wantErr: false},
+
+			// missing HOME env var
+			{path: "~/fizz", wantErr: true},
+		}
+		for i, c := range cases {
+			os.Setenv("HOME", c.home)
+
+			gotPath, err := expandTildePath(c.path)
+			if !c.wantErr && err != nil {
+				t.Error("unexpected error")
+				t.Errorf("case=%d; got=%v; want=%v", i, err, nil)
+			} else if c.wantErr && err == nil {
+				t.Error("unexpected success")
+				t.Error(gotPath)
+				t.Errorf("case=%d; got=%v; want=%v", i, nil, "err")
+			}
+
+			if gotPath != c.wantPath {
+				t.Error("unexpected path")
+				t.Errorf("case=%d; got=%v; want=%v", i, gotPath, c.wantPath)
+			}
+		}
+	})
+
+	t.Run("windows", func(t *testing.T) {
+		if runtime.GOOS != "windows" {
+			t.Skip()
+			return
+		}
+
+		origUserProfile := os.Getenv("USERPROFILE")
+		defer os.Setenv("USERPROFILE", origUserProfile)
+
+		cases := []struct {
+			path string
+			home string
+
+			wantPath string
+			wantErr  bool
+		}{
+			{path: "/Foo/Bar", home: `C:\Foo\Bar`, wantPath: "/Foo/Bar", wantErr: false},
+			{path: "Foo/Bar", home: `C:\Foo\Bar`, wantPath: "Foo/Bar", wantErr: false},
+			{path: "~/Fizz", home: `C:\Foo\Bar`, wantPath: `C:\Foo\Bar\Fizz`, wantErr: false},
+
+			// missing USERPROFILE env var
+			{path: "~/fizz", wantErr: true},
+		}
```

Comment:
Hey, @ColinSullivan1. Are these paths normally how Windows folks would give us a path?

Reply:
Not normally, unless running in some type of non-standard shell.  paths would be \\foo\\bar, wtih backslashes (escaped for the test).